packages/xstate-react/src/createSelector.ts

```diff
@@ -0,0 +1,12 @@
+import { StateFrom, StateMachine } from 'xstate';
+
+/**
+ * Allows you to specify a typesafe selector based on a machine definition.
+ *
+ * @example
+ * const getIsToggedOn = createSelector(machine, state => state.matches('toggledOn'));
+ */
+export const createSelector = <TMachine extends StateMachine<any, any, any>, T>(
+  _machine: TMachine,
+  selector: (state: StateFrom<TMachine>) => T
```

Comment:
We could save TS from resolving this through a helper since we know what the input type is:
```suggestion
  selector: (state: TMachine['initialState']) => T
```